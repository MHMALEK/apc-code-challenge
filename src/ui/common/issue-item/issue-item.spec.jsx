/* eslint-disable jsx-a11y/anchor-is-valid */
import { shallow } from 'enzyme';
import convertIsoStringToDDMMYYYY from 'src/utils/date';
import CommonIssueItem from '.';

describe('CommonItem Componenent render', () => {
  const mockProps = {
    created_at: '2013-08-03T02:00:00Z',
    updated_at: '2013-08-03T02:00:00Z',
    url: 'https://google.com',
    user: 'Malek',
    id: '1',
  };
  const BaseComponent = shallow(<CommonIssueItem {...mockProps} />);

  it('it should match the SnappShot', () => {
    expect(BaseComponent).toMatchSnapshot();
  });
  it('it should renders the base html element', () => {
    expect(BaseComponent.find('a').props().href).toEqual(mockProps.url);
  });
});

describe('CommonItem shoud render props properly', () => {
  const mockProps = {
    created_at: '2013-08-03T02:00:00Z',
    updated_at: '2013-08-03T02:00:00Z',
    url: 'https://google.com',
    user: 'Malek',
    id: '1',
  };

  const convertedCreatedDate = convertIsoStringToDDMMYYYY(mockProps.created_at);
  const convertedUpdatedDate = convertIsoStringToDDMMYYYY(mockProps.updated_at);

  const BaseComponent = shallow(<CommonIssueItem {...mockProps} />);

  it('it should renders the  link element', () => {
    expect(BaseComponent.find('a').props().href).toEqual(mockProps.url);
  });

  it('it should renders the created_at date properly', () => {
    expect(
      BaseComponent.find('p.created_at').html().includes(convertedCreatedDate),
    ).toBeTruthy();
  });

  it('it should renders the created_at date properly', () => {
    expect(
      BaseComponent.find('p.created_at').html().includes(mockProps.created_at),
    ).toBeFalsy();
  });

  it('it should renders the created_at date properly', () => {
    expect(
      BaseComponent.find('p.updated_at').html().includes(convertedUpdatedDate),
    ).toEqual(true);
  });

  it('it should renders the update_at date properly', () => {
    expect(
      BaseComponent.find('p.updated_at').html().includes(mockProps.update_at),
    ).toEqual(false);
  });

  it('it should renders the user data properly', () => {
    expect(
      BaseComponent.find('p.user').html().includes(mockProps.user),
    ).toEqual(true);
  });


});
