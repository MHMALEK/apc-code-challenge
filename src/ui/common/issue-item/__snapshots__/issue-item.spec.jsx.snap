// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CommonItem Componenent render it should match the SnappShot 1`] = `
<div>
  <a
    className="issue-item"
    href="https://google.com"
  >
    <p
      className="user"
    >
      Malek
    </p>
    <p
      className="created_at"
    >
      2013-08-03
    </p>
    <p
      className="updated_at"
    >
      2013-08-03
    </p>
  </a>
</div>
`;
